### PR TITLE
update HistogramObserver to be scriptable

### PR DESCRIPTION
**Pull request by lyoka**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51001 update HistogramObserver to be scriptable**

fix tests in TestQuantizeJitOps

Differential Revision: [D26038759](https://our.internmc.facebook.com/intern/diff/D26038759)

### CODE REVIEW

File: test/quantization/test_workflow_module.py [hunk 1]
```diff
@@ -695,6 +696,37 @@ def test_observer_scriptable(self, qdtype, qscheme):
         loaded = torch.jit.load(buf)
         self.assertTrue(torch.equal(obs.get_tensor_value()[0], loaded.get_tensor_value()[0]))
 
+class TestHistogramObserver(QuantizationTestCase):
```
Comment (raghuramank100) [inline test/quantization/test_workflow_module.py:699]: Wondering if the tests in def test_save_load_state_dict_script(self) already cover this test.

File: torch/quantization/observer.py [hunk 2]
```diff
@@ -697,7 +697,7 @@ class HistogramObserver(_ObserverBase):
     min_val: torch.Tensor
     max_val: torch.Tensor
 
-    def __init__(self, bins=2048, upsample_rate=128, dtype=torch.quint8,
+    def __init__(self, bins: int=2048, upsample_rate=128, dtype=torch.quint8,
```
Comment (raghuramank100) [inline torch/quantization/observer.py:700]: Can we also add dtype annotations for upsample_rate and dtype?

File: torch/quantization/observer.py [hunk 3]
```diff
@@ -58,8 +58,8 @@ def __init__(self, dtype):
         self.dtype = dtype
 
     @abstractmethod
-    def forward(self, x):
-        pass
+    def forward(self, x: torch.Tensor):
```
Comment (raghuramank100) [inline torch/quantization/observer.py:61]: Curious as to why this change is needed for scriptability. Since we override forward, shouldnt that be sufficient?

File: test/quantization/test_workflow_module.py (hunk 1 above)
Comment (lyoka) [reply test/quantization/test_workflow_module.py:699]: that test scripted the model but didn't save to file, and this problem happens at save step.

File: torch/quantization/observer.py (hunk 2 above)
Comment (lyoka) [reply torch/quantization/observer.py]: added.

File: torch/quantization/observer.py (hunk 3 above)
Comment (lyoka) [reply torch/quantization/observer.py]: reverted

**Review by raghuramank100 — APPROVED** [review]
